src/Response/MappedResponseCollection.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace PhpTwinfield\Response;
+
+use Webmozart\Assert\Assert;
+
+class MappedResponseCollection extends \ArrayObject
+{
+    /**
+     * @param mixed $value
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function append($value): void
+    {
+        if (!($value instanceof IndividualMappedResponse)) {
+            throw new \InvalidArgumentException("Value has to be an object of type " . IndividualMappedResponse::class);
+        }
+        parent::append($value);
+    }
+
+    public function hasSuccessfulResponses(): bool
+    {
+        return $this->countResponses(true) > 0;
+    }
+
+    public function countSuccessfulResponses(): int
+    {
+        return $this->countResponses(true);
+    }
+
+    /**
+     * @return IndividualMappedResponse[]
+     */
+    public function getSuccessfulResponses(): array
+    {
+        return $this->getResponses(true);
+    }
+
+    public function hasFailedResponses(): bool
+    {
+        return $this->countResponses(false) > 0;
+    }
+
+    public function countFailedResponses(): int
+    {
+        return $this->countResponses(false);
+    }
+
+    /**
+     * @return IndividualMappedResponse[]
+     */
+    public function getFailedResponses(): array
+    {
+        return $this->getResponses(false);
+    }
+
+    public function assertAllSuccessful(): void
```

Comment:
Could you add an `@throws InvalidArgumentException` here, since that is the exception that is thrown when the assertion fails. That way the caller knows how to handle this.

Reply:
This should actually have its own exception class, since it is not an invalid argument, but it won't matter that much.